src/Form/RedirectFix404Form.php

```diff
@@ -55,51 +55,63 @@ public function buildForm(array $form, FormStateInterface $form_state) {
+    }
+    $header[] = array('data' => t('Operations'));
 
-    $count_query = db_select('watchdog', 'w');
-    $count_query->addExpression('COUNT(DISTINCT(w.message))');
-    $count_query->leftJoin('redirect', 'r', 'w.message = r.redirect_source__path');
-    $count_query->condition('w.type', 'page not found');
+    $count_query = db_select('redirect_error', 're');
+    $count_query->addExpression('COUNT(DISTINCT(re.source))');
+    $count_query->leftJoin('redirect', 'r', 're.source = r.redirect_source__path');
     $count_query->isNull('r.rid');
-    $this->filterQuery($count_query, array('w.message'), $search);
+    $this->filterQuery($count_query, array('re.source'), $search);
 
-    $query = db_select('watchdog', 'w')
+    $query = db_select('redirect_error', 're')
       ->extend('Drupal\Core\Database\Query\TableSortExtender')->orderByHeader($header)
       ->extend('Drupal\Core\Database\Query\PagerSelectExtender')->limit(25);
-    $query->fields('w', array('message'));
-    $query->addExpression('COUNT(wid)', 'count');
+    $query->fields('re', array('source', 'language'));
+    $query->addExpression('COUNT(reid)', 'count');
     $query->addExpression('MAX(timestamp)', 'timestamp');
-    $query->leftJoin('redirect', 'r', 'w.message = r.redirect_source__path');
+    $query->leftJoin('redirect', 'r', 're.source = r.redirect_source__path');
     $query->isNull('r.rid');
-    $query->condition('w.type', 'page not found');
-    $query->groupBy('w.message');
-    $this->filterQuery($query, array('w.message'), $search);
+    $query->groupBy('re.source, re.language');
+    $this->filterQuery($query, array('re.source'), $search);
     $query->setCountQuery($count_query);
     $results = $query->execute();
 
     $rows = array();
     foreach ($results as $result) {
 
-      // @todo Detect the language from the url.
-      $request = Request::create($result->message, 'GET', [], [], [], \Drupal::request()->server->all());
+      $request = Request::create($result->source, 'GET', [], [], [], \Drupal::request()->server->all());
       $path = ltrim($request->getPathInfo(), '/');
 
       $row = array();
-      $row['source'] = \Drupal::l($result->message, Url::fromUri('base:' . $path, array('query' => $destination)));
+      $row['source'] = \Drupal::l($result->source, Url::fromUri('base:' . $path, array('query' => $destination)));
       $row['count'] = $result->count;
       $row['timestamp'] = format_date($result->timestamp, 'short');
+      if ($multilanguage) {
+        $row['language'] = $result->language;
+        if ($result->language == '' || isset($languages[$result->language])) {
+          $row['language'] = $result->language == '' ? t('Language neutral') : t('@lang_name', array('@lang_name' => $languages[$result->language]->getName()));
+        }
+        else {
+          $row['language'] = t('Undefined language (@langcode)', array('@langcode' => $result->language));
```

Comment:
That "Undefined language..." will make the column explode. Reduce it to something like "Undefined @langcode".